File: SimpleMoxieSwitcher/Sources/SimpleMoxieSwitcher/ViewModels/AppearanceViewModel.swift
import SwiftUI
import Combine

@MainActor
class AppearanceViewModel: ObservableObject {
    @Published var isApplying = false
    @Published var statusMessage: String?

    private let appearanceService: AppearanceServiceProtocol
    private let mqttService: MQTTServiceProtocol

    init(appearanceService: AppearanceServiceProtocol, mqttService: MQTTServiceProtocol) {
        self.appearanceService = appearanceService
        self.mqttService = mqttService
    }

    func applyAppearance(
        eyes: String,
        faceColors: String,
        eyeDesigns: String,
        faceDesigns: String,
        eyelidDesigns: String,
        mouth: String,
        headHair: String,
        facialHair: String,
        brows: String,
        glasses: String,
        nose: String
    ) async {
        isApplying = true
        statusMessage = "Applying appearance changes..."

        let appearance = AppearanceSettings(
            eyes: eyes,
            faceColors: faceColors,
            eyeDesigns: eyeDesigns,
            faceDesigns: faceDesigns,
            eyelidDesigns: eyelidDesigns,
            mouth: mouth,
            headHair: headHair,
            facialHair: facialHair,
            brows: brows,
            glasses: glasses,
            nose: nose
        )

        await appearanceService.applyAppearance(appearance)

        isApplying = false
        statusMessage = "Appearance updated successfully!"
    }
}

// MARK: - Appearance Settings Model
struct AppearanceSettings {
    let eyes: String
    let faceColors: String
    let eyeDesigns: String
    let faceDesigns: String
    let eyelidDesigns: String
    let mouth: String
    let headHair: String
    let facialHair: String
    let brows: String
    let glasses: String
    let nose: String
}

File: SimpleMoxieSwitcher-Windows/SimpleMoxieSwitcher/ViewModels/AppearanceViewModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Runtime.CompilerServices;
using System.Threading.Tasks;
using SimpleMoxieSwitcher.Models;
using SimpleMoxieSwitcher.Services;

namespace SimpleMoxieSwitcher.ViewModels
{
    /// <summary>
    /// ViewModel for customizing Moxie's appearance
    /// </summary>
    public class AppearanceViewModel : INotifyPropertyChanged
    {
        private bool _isApplying;
        private string? _statusMessage;

        private readonly IAppearanceService _appearanceService;
        private readonly IMQTTService _mqttService;

        public AppearanceViewModel(IAppearanceService appearanceService, IMQTTService mqttService)
        {
            _appearanceService = appearanceService;
            _mqttService = mqttService;
        }

        #region Properties

        public bool IsApplying
        {
            get => _isApplying;
            set => SetProperty(ref _isApplying, value);
        }

        public string? StatusMessage
        {
            get => _statusMessage;
            set => SetProperty(ref _statusMessage, value);
        }

        #endregion

        #region Methods

        public async Task ApplyAppearanceAsync(
            string eyes,
            string faceColors,
            string eyeDesigns,
            string faceDesigns,
            string eyelidDesigns,
            string mouth,
            string headHair,
            string facialHair,
            string brows,
            string glasses,
            string nose)
        {
            IsApplying = true;
            StatusMessage = "Applying appearance changes...";

            var appearance = new AppearanceSettings
            {
                Eyes = eyes,
                FaceColors = faceColors,
                EyeDesigns = eyeDesigns,
                FaceDesigns = faceDesigns,
                EyelidDesigns = eyelidDesigns,
                Mouth = mouth,
                HeadHair = headHair,
                FacialHair = facialHair,
                Brows = brows,
                Glasses = glasses,
                Nose = nose
            };

            await _appearanceService.ApplyAppearanceAsync(appearance);

            IsApplying = false;
            StatusMessage = "Appearance updated successfully!";
        }

        #endregion

        #region INotifyPropertyChanged

        public event PropertyChangedEventHandler? PropertyChanged;

        protected virtual void OnPropertyChanged([CallerMemberName] string? propertyName = null)
        {
            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
        }

        protected bool SetProperty<T>(ref T field, T value, [CallerMemberName] string? propertyName = null)
        {
            if (EqualityComparer<T>.Default.Equals(field, value))
                return false;

            field = value;
            OnPropertyChanged(propertyName);
            return true;
        }

        #endregion
    }

    #region Supporting Models

    /// <summary>
    /// Settings for customizing Moxie's appearance
    /// </summary>
    public class AppearanceSettings
    {
        public string Eyes { get; set; } = "";
        public string FaceColors { get; set; } = "";
        public string EyeDesigns { get; set; } = "";
        public string FaceDesigns { get; set; } = "";
        public string EyelidDesigns { get; set; } = "";
        public string Mouth { get; set; } = "";
        public string HeadHair { get; set; } = "";
        public string FacialHair { get; set; } = "";
        public string Brows { get; set; } = "";
        public string Glasses { get; set; } = "";
        public string Nose { get; set; } = "";
    }

    #endregion
}
